paymentsheet/src/test/java/com/stripe/android/paymentsheet/repositories/CustomerRepositoryTest.kt

```diff
@@ -171,6 +171,66 @@ internal class CustomerRepositoryTest {
             assertThat(result[0].id).isEqualTo("pm_1")
         }
 
+    @Test
+    fun `getPaymentsMethods() should keep cards from Link wallet`() = runTest {
+        givenGetPaymentMethodsReturns(
+            Result.success(emptyList())
+        )
+
+        val initialCard = PaymentMethodFixtures.CARD_PAYMENT_METHOD
+
+        val mockedReturnPaymentMethods = listOf(
+            initialCard.copy("pm_1"),
+            initialCard.copy(
+                id = "pm_2",
+                card = initialCard.card?.copy(
+                    wallet = Wallet.GooglePayWallet("3000")
+                )
+            ),
+            initialCard.copy(
+                id = "pm_3",
+                card = initialCard.card?.copy(
+                    wallet = Wallet.LinkWallet("4000")
+                )
+            ),
+            initialCard.copy(
+                id = "pm_4",
+                card = initialCard.card?.copy(
+                    wallet = Wallet.LinkWallet("5000")
+                )
+            ),
+        )
+
+        stripeRepository.stub {
+            onBlocking {
+                getPaymentMethods(
+                    listPaymentMethodsParams = eq(
+                        ListPaymentMethodsParams(
+                            customerId = "customer_id",
+                            paymentMethodType = PaymentMethod.Type.Card
```

Comment:
What happens for us_bank_account? Are link accounts returned there? Does everything work?

Reply:
They are returned there but as a `PaymentMethod` object with only the `link` attribute filled in (see [here](https://docs.stripe.com/api/payment_methods/object#payment_method_object-link)). There's no way to dedupe this since we only have the email so we will just show all of them.

We've opted not to show them the US bank account ones because of the limited data on iOS.

Reply:
This is being filtered in the `PaymentSheetLoader` classes.

Reply:
if there isn't already a test for this, can you add one as a follow up?